src/services/dimension-processor.ts

```diff
@@ -310,14 +229,58 @@ async function recreateBaseFactTable(dataset: Dataset, dataTable: DataTable): Pr
     }
 }
 
+async function removeAllDimensions(dataset: Dataset) {
+    logger.warn(`Removing all dimensions for dataset ${dataset.id}`);
+    for (const dimension of dataset.dimensions) {
+        if (dimension.lookupTable) {
+            const dataLakeService = new DataLakeService();
+            try {
+                dataLakeService.deleteFile(dimension.lookupTable.filename, dataset.id);
+            } catch (error) {
+                logger.warn(
+                    error,
+                    `Something went wrong trying to remove previously uploaded lookup table with error: ${error}`
+                );
+            }
+            logger.warn(`Removing lookup table for dimension ${dimension.id}`);
+            await LookupTable.getRepository().delete({ dimension });
+        }
+        await DimensionMetadata.getRepository().delete({ dimension });
+    }
+    await Dimension.getRepository().delete({ dataset });
+}
+
+async function removeMeasure(dataset: Dataset) {
+    logger.warn(`Removing measure for dataset ${dataset.id}`);
+    if (dataset.measure) {
+        if (dataset.measure.lookupTable) {
+            const dataLakeService = new DataLakeService();
+            try {
+                dataLakeService.deleteFile(dataset.measure.lookupTable.filename, dataset.id);
+            } catch (error) {
+                logger.warn(
+                    error,
+                    `Something went wrong trying to remove previously uploaded lookup table with error: ${error}`
+                );
+            }
+            logger.warn(`Removing lookup table for measure ${dataset.measure.id}`);
+            await LookupTable.getRepository().delete({ measure: dataset.measure });
+        }
+        await MeasureRow.getRepository().delete({ measure: dataset.measure });
+        await MeasureMetadata.getRepository().delete({ measure: dataset.measure });
```

Comment:
Similar to above - both MeasureRow and MeasureMetadata have an `onDelete:'CASCADE'` on the measure relation - shouldn't this get triggered automatically by deleting the measure?